src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php

```diff
@@ -149,14 +149,43 @@ public function addData(array $indexData, $storeId)
         return $productsList;
     }
 
+    /**
+     * @param array $indexData
+     * @param $storeId
+     * @param \Divante\VsbridgeIndexerCatalog\Model\Indexer\DataProvider\Product\ConfigurableData $dataProvider
+     *
+     * @return array
+     * @throws \Exception
+     */
+    public function addDataWithMediaGallery(array $indexData, $storeId, \Divante\VsbridgeIndexerCatalog\Model\Indexer\DataProvider\Product\MediaGalleryData $dataProviderMG)
```

Comment:
Only
1. `addData` function is allowed to add data, this code should be moved to `addData`  function.
2. It should be configurable in panel to add images for children or not - so we have more flexible solution
MediaGallery Provider should be refactor to extract logic, so we can use it in multiple places.

We will be working on more flexible soluton in #105 - I won't be merging this changes.